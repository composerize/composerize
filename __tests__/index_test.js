/* eslint-env jest */

import Composerize from '../src';

it('fails with invalid commands', () => {
    expect(() => {
        Composerize('foo bar');
    }).toThrow();
});

describe('snapshots', () => {
    const CMDS = [
        'docker run -p 80:80 -v /var/run/docker.sock:/tmp/docker.sock:ro --restart always --log-opt max-size=1g nginx',

        // test spacing
        ' docker   run -p 80:80  -v /var/run/docker.sock:/tmp/docker.sock:ro --restart always --log-opt max-size=1g nginx    ',

        // test multiple args (https://github.com/magicmark/composerize/issues/9)
        'docker run -t --name="youtrack" -v /data/youtrack/data/:/opt/youtrack/data/ -v /data/youtrack/backup/:/opt/youtrack/backup/ -p 80:80 -p 3232:22351 uniplug/youtrack',

        // testing parsing of quotes (https://github.com/magicmark/composerize/issues/10)
        'docker run --name="foobar" nginx',

        'docker run -p 80:80 --name webserver nginx:latest',

        'docker run -p 80:80 foobar/baz:latest',

        // test flag
        'docker run --privileged -p 80:80 foobar/baz:latest',

        // https://github.com/magicmark/composerize/issues/25
        'docker run -d --name plex --network=host -e TZ="<timezone>" -e PLEX_CLAIM="<claimToken>" -v <path/to/plex/database>:/config -v <path/to/transcode/temp>:/transcode -v <path/to/media>:/data plexinc/pms-docker',
      
        'docker run -d --name plex --net=host -e TZ="<timezone>" -e PLEX_CLAIM="<claimToken>" -v <path/to/plex/database>:/config -v <path/to/transcode/temp>:/transcode -v <path/to/media>:/data plexinc/pms-docker',
      
        // test create
        'docker create -p 80:80 --name webserver nginx:latest',
      
        //test pid
        'docker run -p 80:80 --pid="host" --name webserver nginx:latest',
      
        'docker run -p 80:80 --pid="host" --ulimit nofile=1234:1234 --name webserver nginx:latest'
    ];

    it('match snapshots', () => {
        CMDS.forEach(cmd => {
            expect(Composerize(cmd)).toMatchSnapshot();
        });
    });
});
