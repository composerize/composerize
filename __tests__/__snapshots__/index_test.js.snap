// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots match snapshots 1`] = `
"version: '3.3'
services:
    nginx:
        ports:
            - '80:80'
        volumes:
            - '/var/run/docker.sock:/tmp/docker.sock:ro'
        restart: always
        logging:
            options:
                max-size: 1g
        image: nginx"
`;

exports[`snapshots match snapshots 2`] = `
"version: '3.3'
services:
    nginx:
        ports:
            - '80:80'
        volumes:
            - '/var/run/docker.sock:/tmp/docker.sock:ro'
        restart: always
        logging:
            options:
                max-size: 1g
        image: nginx"
`;

exports[`snapshots match snapshots 3`] = `
"version: '3.3'
services:
    youtrack:
        container_name: youtrack
        volumes:
            - '/data/youtrack/data/:/opt/youtrack/data/'
            - '/data/youtrack/backup/:/opt/youtrack/backup/'
        ports:
            - '80:80'
            - '3232:22351'
        image: uniplug/youtrack"
`;

exports[`snapshots match snapshots 4`] = `
"version: '3.3'
services:
    nginx:
        container_name: foobar
        image: nginx"
`;

exports[`snapshots match snapshots 5`] = `
"version: '3.3'
services:
    nginx:
        ports:
            - '80:80'
        container_name: webserver
        image: 'nginx:latest'"
`;

exports[`snapshots match snapshots 6`] = `
"version: '3.3'
services:
    baz:
        ports:
            - '80:80'
        image: 'foobar/baz:latest'"
`;

exports[`snapshots match snapshots 7`] = `
"version: '3.3'
services:
    baz:
        privileged: true
        ports:
            - '80:80'
        image: 'foobar/baz:latest'"
`;

exports[`snapshots match snapshots 8`] = `
"version: '3.3'
services:
    pms-docker:
        container_name: plex
        network_mode: host
        environment:
            - TZ=<timezone>
            - PLEX_CLAIM=<claimToken>
        volumes:
            - '<path/to/plex/database>:/config'
            - '<path/to/transcode/temp>:/transcode'
            - '<path/to/media>:/data'
        image: plexinc/pms-docker"
`;

exports[`snapshots match snapshots 9`] = `
"version: '3.3'
services:
    pms-docker:
        container_name: plex
        network_mode: host
        environment:
            - TZ=<timezone>
            - PLEX_CLAIM=<claimToken>
        volumes:
            - '<path/to/plex/database>:/config'
            - '<path/to/transcode/temp>:/transcode'
            - '<path/to/media>:/data'
        image: plexinc/pms-docker"
`;

exports[`snapshots match snapshots 10`] = `
"version: '3.3'
services:
    nginx:
        ports:
            - '80:80'
        container_name: webserver
        image: 'nginx:latest'"
`;

exports[`snapshots match snapshots 11`] = `
"version: '3.3'
services:
    nginx:
        ports:
            - '80:80'
        pid: host
        container_name: webserver
        image: 'nginx:latest'"
`;

exports[`snapshots match snapshots 12`] = `
"version: '3.3'
services:
    nginx:
        ports:
            - '80:80'
        ulimits:
          nofile:
            soft: 1234
            hard: 1234
        pid: host
        container_name: webserver
        image: 'nginx:latest'"
`;
